src/Sylius/Behat/Context/Api/Shop/CartContext.php

```diff
@@ -742,31 +742,33 @@ public function iShouldBeUnableToAddItToTheCart(): void
     }
 
     /**
-     * @Then /^(this product) should have ([^"]+) "([^"]+)"$/
+     * @Then /^this product should have ([^"]+) "([^"]+)"$/
      */
-    public function thisItemShouldHaveOptionValue(ProductInterface $product, string $optionName, string $optionValue): void
+    public function thisItemShouldHaveOptionValue(string $expectedOptionName, string $expectedOptionValueValue): void
     {
         $item = $this->sharedStorage->get('item');
 
-        $variantData = json_decode($this->shopClient->showByIri(urldecode($item['variant']))->getContent(), true, 512, \JSON_THROW_ON_ERROR);
+        $variantData = $this->responseChecker->getResponseContent($this->shopClient->showByIri($item['variant']));
```

Comment:
```suggestion
        $optionValues = $this->responseChecker->getValue($this->shopClient->showByIri($item['variant']), 'optionValues');
```
similar in the calls below